yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,264 @@
+		//which weren't present the starting and during the run it came back.
+		if !slices.Contains(lo.Keys(m.nodesStatus), conf.Host) {
+			m.nodesStatus[conf.Host] = true
+		}
+	}
+
+	if len(downNodes) > 0 {
+		downNodeMsg := color.RedString(fmt.Sprintf("ALERT: %s. Unhealthy nodes: %v\n", NODE_GOES_DOWN_MSG, strings.Join(downNodes, ", ")))
+		m.displayMsgFunc(downNodeMsg)
+		err := m.yb.RemoveConnectionsForHosts(downNodes)
+		if err != nil {
+			return fmt.Errorf("error while removing the connections for the down nodes[%v]: %v", downNodes, err)
+		}
+		return nil
+	}
+	if len(upNodes) > 0 {
+		upNodeMsg := color.GreenString(fmt.Sprintf("OK: %s. Recovered nodes: %v\n", NODE_COMES_BACK_UP_MSG, strings.Join(upNodes, ", ")))
+		m.displayMsgFunc(upNodeMsg)
+	}
+	return nil
+}
+
+/*
+If disk space on any node is less than 10%, error out the import and let the user increase the disk
+
+//TODO: enhance yb_server_metrics() to give these metrics.
+*/
+func (m *MonitorTargetYBHealth) monitorDiskUsageAndAbort() error {
+	if m.skipDiskCheck {
+		return nil
+	}
+	nodeMetrics, err := m.yb.GetClusterMetrics()
+	if err != nil {
+		return fmt.Errorf("error fetching cluster metrics: %v", err)
+	}
+	var totalFreeDisk, totalDisk int64
+	for _, metrics := range nodeMetrics {
+		if metrics.Status != "OK" {
+			continue
+		}
+		nodeFreeDisk, err := strconv.ParseInt(metrics.Metrics[METRIC_FREE_DISK_SPACE], 10, 64)
+		if err != nil {
+			return fmt.Errorf("parsing free disk space metric as int: %w", err)
+		}
+		nodeTotalDisk, err := strconv.ParseInt(metrics.Metrics[METRIC_TOTAL_DISK_SPACE], 10, 64)
+		if err != nil {
+			return fmt.Errorf("parsing total disk space metric as int: %w", err)
+		}
+		totalFreeDisk += nodeFreeDisk
+		totalDisk += nodeTotalDisk
+		freeDiskPct := (nodeFreeDisk * 100) / nodeTotalDisk
+		if freeDiskPct < FREE_DISK_SPACE_THREASHOLD {
+			utils.ErrExit("Free disk space available on the target cluster is low, increase the disk space on the target cluster and re-run the import.")
+		}
+	}
+	return nil
+}
+
+/*
+If any of the replication streams are enabled on the cluster during the import -
+let the user know to enable it after import data is done to avoid wal or disk space issues because of replication
+
+Caveats -
+1. For all deployments - only logical replication info is easily detectable - using the pg_replciation_slots for getting num of logical replication slots
+2. For YBA/yugabyted - with above also trying to figure the xcluster replication streams/CDC gRPC streams using the yb-client wrapper jar to fetch num of all cdc streams on the cluster
+*/
+func (m *MonitorTargetYBHealth) monitorReplicationOnTarget() error {
+	if m.skipReplicationCheck {
+		return nil
+	}
+	numOfSlots, err := m.yb.NumOfLogicalReplicationSlots()
+	if err != nil {
+		return fmt.Errorf("error fetching logical replication slots for checking if replication enabled - %s", err)
+	}
+	if numOfSlots > 0 {
+		utils.ErrExit(color.RedString(REPLICATION_GUARDRAIL_ALERT_MSG))
```

Comment:
We should mention in the error msg : "we found x no. of replication slots"
same for the replication streams case.